BinaryEncoding.md

```diff
@@ -786,9 +783,138 @@ for [future :unicorn:][future multiple tables] use and must be 0 in the MVP.
+| `i64.atomic_sub8_u` | `0xe617` | `1` `memory immediate` | atomic sub from memory |
+| `i64.atomic_sub16_s` | `0xe618` | `1` `memory immediate` | atomic sub from memory |
+| `i64.atomic_sub16_u` | `0xe619` | `1` `memory immediate` | atomic sub from memory |
+| `i64.atomic_sub32_s` | `0xe61a` | `1` `memory immediate` | atomic sub from memory |
+| `i64.atomic_sub32_u` | `0xe61b` | `1` `memory immediate` | atomic sub from memory |
+| `i32.atomic_and` | `0xe610` | `2` `memory immediate` | atomic and with memory |
+| `i64.atomic_and` | `0xe611` | `2` `memory immediate` | atomic and with memory |
+| `i32.atomic_and8_s` | `0xe612` | `2` `memory immediate` | atomic and with memory |
+| `i32.atomic_and8_u` | `0xe613` | `2` `memory immediate` | atomic and with memory |
+| `i32.atomic_and16_s` | `0xe614` | `2` `memory immediate` | atomic and with memory |
+| `i32.atomic_and16_u` | `0xe615` | `2` `memory immediate` | atomic and with memory |
+| `i64.atomic_and8_s` | `0xe616` | `2` `memory immediate` | atomic and with memory |
+| `i64.atomic_and8_u` | `0xe617` | `2` `memory immediate` | atomic and with memory |
+| `i64.atomic_and16_s` | `0xe618` | `2` `memory immediate` | atomic and with memory |
+| `i64.atomic_and16_u` | `0xe619` | `2` `memory immediate` | atomic and with memory |
+| `i64.atomic_and32_s` | `0xe61a` | `2` `memory immediate` | atomic and with memory |
+| `i64.atomic_and32_u` | `0xe61b` | `2` `memory immediate` | atomic and with memory |
+| `i32.atomic_or` | `0xe610` | `3` `memory immediate` | atomic or with memory |
+| `i64.atomic_or` | `0xe611` | `3` `memory immediate` | atomic or with memory |
+| `i32.atomic_or8_s` | `0xe612` | `3` `memory immediate` | atomic or with memory |
+| `i32.atomic_or8_u` | `0xe613` | `3` `memory immediate` | atomic or with memory |
+| `i32.atomic_or16_s` | `0xe614` | `3` `memory immediate` | atomic or with memory |
+| `i32.atomic_or16_u` | `0xe615` | `3` `memory immediate` | atomic or with memory |
+| `i64.atomic_or8_s` | `0xe616` | `3` `memory immediate` | atomic or with memory |
+| `i64.atomic_or8_u` | `0xe617` | `3` `memory immediate` | atomic or with memory |
+| `i64.atomic_or16_s` | `0xe618` | `3` `memory immediate` | atomic or with memory |
+| `i64.atomic_or16_u` | `0xe619` | `3` `memory immediate` | atomic or with memory |
+| `i64.atomic_or32_s` | `0xe61a` | `3` `memory immediate` | atomic or with memory |
+| `i64.atomic_or32_u` | `0xe61b` | `3` `memory immediate` | atomic or with memory |
+| `i32.atomic_xor` | `0xe610` | `4` `memory immediate` | atomic xor with memory |
+| `i64.atomic_xor` | `0xe611` | `4` `memory immediate` | atomic xor with memory |
+| `i32.atomic_xor8_s` | `0xe612` | `4` `memory immediate` | atomic xor with memory |
+| `i32.atomic_xor8_u` | `0xe613` | `4` `memory immediate` | atomic xor with memory |
+| `i32.atomic_xor16_s` | `0xe614` | `4` `memory immediate` | atomic xor with memory |
+| `i32.atomic_xor16_u` | `0xe615` | `4` `memory immediate` | atomic xor with memory |
```

Comment:
Names like `i32.atomic_xor16_s` are a little awkward with `_s`/`_u` apparently associated with otherwise sign-agnostic operators. What would you think of any of the following conventions?

 - `i32.atomic16_s.xor`
 - `i32.atomic_rmw16_s.xor`
 - `i32.atomic.rmw16_s.xor`   (in this case, following atomic with `.` could be a broader convention)

These would make it clear that the xor isn't signed itself; it's just the loaded value from the rmw that's being sign-extended. This isn't super important with the current opcodes, but could be in the future if wasm ever wants atomic rotate, popcnt, or other operators which care about the number of bits in the actual arithmetic part.

Such a clarification would also align with an idea anticipated [here](https://github.com/WebAssembly/design/blob/master/FutureFeatures.md#integer-overflow-detection) concerning the possibility of adding a real `add_s` and similar operators (which could trap or do other things on overflow) in the future.